src/soc/intel/baytrail/Makefile.mk

```diff
@@ -87,33 +93,26 @@ endif
 
 ifeq ($(CONFIG_TXE_SECURE_BOOT),y)
 
-cbfs-files-y += manifests.bin
-manifests.bin-file := $(objcbfs)/sb_manifests
-manifests.bin-type := raw
-
-ifeq ($(CONFIG_TXE_SB_INCLUDE_KEY_MANIFEST),y)
+# Add BSP microcode to bootblock region
+microcode_pos = $(call int-subtract, $(CONFIG_ROM_SIZE) $(call int-subtract, 0x100000000 $(CONFIG_EARLY_CPU_MICROCODE_CBFS_LOC)))
 
-manifests.bin-COREBOOT-position := $(call int-subtract, 0x100000000 0x21000)
+add_bootblock = \
+	$(CBFSTOOL) $(1) write -d -r BOOTBLOCK -f $(2); \
+	dd if=$(cpu_microcode_blob.bin-file) of=$(1) conv=notrunc bs=1 \
+		seek=$(microcode_pos) >/dev/null 2>&1
 
+ifeq ($(CONFIG_TXE_SB_INCLUDE_KEY_MANIFEST),y)
 ifneq ($(call strip_quotes,$(CONFIG_TXE_SB_KEY_MANIFEST_PATH)),)
 
-$(objcbfs)/sb_manifests: $(call strip_quotes,$(CONFIG_TXE_SB_KEY_MANIFEST_PATH))
-	dd if=/dev/zero of=$@ bs=5120 count=1 2> /dev/null
-	dd if=$< of=$@ conv=notrunc 2> /dev/null
-else
+key_manifest_pos = $(call int-subtract, $(CONFIG_ROM_SIZE) 0x21000)
```

Comment:
I'm not sure I get it. What is wrong with the hardcoded size?

Reply:
Ohh I get it, you mean to simply put the manifest using cbfstool without defining the lcoation.

Reply:
@krystian-hebel so should we leave the hardcoded size?

Reply:
> MANIFESTS region may be placed anywhere...

Then it would be better to assert that this region is properly located, than blindly `dd` into offset that may be outside of MANIFESTS.

Reply:
> Then it would be better to assert that this region is properly located, than blindly dd into offset that may be outside of MANIFESTS.

Yes, it could be checked somewhere. But blind DD is better than building an image which won't get the CPU out of reset...

Reply:
Check implemented: a52993b1b61e111db94fa1b35574d9e7565a72a7
Also writing the key manifest with cbfstool using MANIFESTS region.

Reply:
It works, but is there a reason why there are two commas in `$(filter)`?

Reply:
4a0e6c1f7e43e90fb061c1e1d924d4cc234fccce

No reason, probably some leftover from previous attempt to get this logic working...